src/test/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocMethodCheckTest.java

```diff
@@ -595,4 +598,32 @@ public void testWithoutLogErrors() throws Exception {
                 CommonUtil.EMPTY_STRING_ARRAY);
     }
 
+    @Test
+    public void testClassAliasResolving() throws Exception {
```

Comment:
I am still looking over all discussions but I wanted to point to https://github.com/checkstyle/checkstyle/issues/7980#issuecomment-607310043 . If reflection stays, we need javadoc on why reflection is needed.